src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,628 @@
+          )}ing</div>
+          <div class="modes">
+            <ha-icon
+              class="${
+                stateObj.attributes.operation_mode === "auto"
+                  ? "selected-icon"
+                  : ""
+              }"
+              entity="${stateObj.entity_id}"
+              id="auto"
+              @click="${this.handleModeClick}"
+              icon="${modeIcons.auto}"
+            ></ha-icon>
+            <ha-icon
+              class="${
+                stateObj.attributes.operation_mode === "heat"
+                  ? "selected-icon"
+                  : ""
+              }"
+              id="heat"
+              @click="${this.handleModeClick}"
+              icon="${modeIcons.heat}"></ha-icon>
+            <ha-icon
+              class="${
+                stateObj.attributes.operation_mode === "cool"
+                  ? "selected-icon"
+                  : ""
+              }"
+              id="cool"
+              @click="${this.handleModeClick}"
+              icon="${modeIcons.cool}"></ha-icon>
+            <ha-icon
+              class="${
+                stateObj.attributes.operation_mode === "off"
+                  ? "selected-icon"
+                  : ""
+              }"
+              id="off"
+              @click="${this.handleModeClick}"
+              icon="${modeIcons.off}"></ha-icon>
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected updated(changedProperties) {
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    $("#thermostat", this.shadowRoot).roundSlider({
+      radius: this.clientWidth / 3,
+      min: stateObj.attributes.min_temp,
+      max: stateObj.attributes.max_temp,
+      step: thermostatConfig.step,
+      circleShape: thermostatConfig.circleShape,
+      sliderType: thermostatConfig.sliderType,
+      startAngle: thermostatConfig.startAngle,
+      width: thermostatConfig.width,
+      handleSize: thermostatConfig.handleSize,
+      lineCap: thermostatConfig.lineCap,
+      showTooltip: thermostatConfig.showTooltip,
+      editableTooltip: false,
+      value: stateObj.attributes.temperature,
+      tooltipFormat: this.tooltip.bind(this),
+      change: this.setTemperature.bind(this),
+    });
+  }
+
+  private renderStyle() {
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    return html`
+    <style>
+
+    ha-card {
+      overflow: hidden;
+    }
+
+    .modes {
+      padding-top: 5px;
+    }
+
+    .title {
+      font-size: ${Math.round($(this).outerWidth() / 18)}px;
+      padding-bottom: 40px;
+    }
+
+    .climate-info {
+      position: absolute;
+      bottom: -3%;
+      left: 50%;
+      transform: translate(-50%,-50%);
+      z-index: 5;
+      text-align: center;
+    }
+
+    .modes ha-icon {
+      color: var(--disabled-text-color);
+      cursor: pointer;
+      padding: 0 10px;
+    }
+
+    .modes .selected-icon {
+      color: ${modeColor[stateObj.attributes.operation_mode]};
```

Comment:
operation mode should be applied as a class to the object and we should keep our CSS static.